src/tests/signup.test.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable mocha/no-hooks-for-single-case */
+/* eslint-disable no-unused-expressions */
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import cookieParser from 'cookie-parser';
+import bcrypt from 'bcrypt';
+import server from '../index.js';
+import User from '../models/user.model.js';
+
+const { hash } = bcrypt;
+
+chai.should();
+chai.use(chaiHttp);
+chai.use(cookieParser);
+
+describe('Testing Signup Route with errors', function () {
+  before(async function () {
+    const password = await hash('Qwert@12345', 10);
+    const testUser = {
+      email: 'testish111@mail.com',
+      username: 'testish111',
+      password,
+    };
+    await User.create(testUser);
+  });
+  after(async function () {
+    const user = await User.findOne({ where: { username: 'testish111' } });
+    await User.destroy({ where: { id: user.dataValues.id } });
+  });
+  it('should not signup successfully if email already exists', async function () {
+    // Create a user with a unique ID
+    const user = {
+      email: 'testish111@mail.com',
+      username: 'testish112',
+      password: 'Qwert@12345',
+    };
+    // Log in as the user and get the session ID
+    const agent = chai.request.agent(server);
+    const response = await agent.post('/users/signup').send({
+      email: user.email,
+      username: user.username,
+      password: user.password,
+    });
+    // // Expect the response to have a status code of 201 and token
+    chai.expect(response).to.have.status(400);
+    chai
+      .expect(response.body)
+      .to.be.an('object')
+      .to.have.property('error', 'email already exists.');
+  });
+  it('should not signup successfully if username already exists', async function () {
```

Comment:
This test will fail plz! Remember that @AdrineUWERA  has changed username, now it is no longer unique! so existence of username can not cause a problem, that is why you will get 201 instead of 400. check well, or delete this test! bcz it is no longer useful unless the username is unique. 
## Remember
### This is working in your tests but after merging it will fail.

Reply:
Resolved